File: src/main/java/com/snowofsunflower/android/ui/list/IRecyclerViewListAdapter.java
package com.snowofsunflower.android.ui.list;

import android.support.v7.widget.RecyclerView;

import java.util.List;

/**
 * Created by zhouztashin on 2018/10/31.
 */

public interface IRecyclerViewListAdapter<T> {


    RecyclerView.Adapter getAdapter();
    int getItemLayoutId();
    void doItem(ItemViewHolder itemViewHolder, T item);
    void setNewData(List<T> list);
    T getItem(int position);
    void setOnItemClickListener(OnItemClickListener listener);
}

File: src/main/java/com/snowofsunflower/android/ui/list/IRecyclerViewListAdapter.kt
package com.snowofsunflower.android.ui.list

import androidx.recyclerview.widget.RecyclerView

/**
 * Created by zhouztashin on 2018/10/31.
 * 列表适配器
 */

interface IRecyclerViewListAdapter<T> {


    val adapter: RecyclerView.Adapter<*>
    val itemLayoutId: Int
    fun doItem(itemViewHolder: ItemViewHolder, item: T)
    fun setNewData(list: List<T>)
    fun getItem(position: Int): T?
    fun <T> setOnItemClickListener(listener: (SimpleListAdapter<*>, Int, T) -> Unit)
}
